### PR TITLE
bump NavNative to 116.0.0, Common to 23.1.0-beta.2 and Maps to 10.9.0-beta.2

### PR DESCRIPTION
**Pull request by Zayankovsky**

### CODE REVIEW

**Review by Guardiola31337 — APPROVED** [review]

**Comment by Zayankovsky** [comment]
Android Auto crashes because of a compatibility in Maps SDK:
```
FATAL EXCEPTION: main
Process: com.mapbox.navigation.qa_test_app, PID: 25740
java.lang.NoSuchMethodError: No direct method <init>(Landroid/graphics/drawable/Drawable;Landroid/graphics/drawable/Drawable;Landroid/graphics/drawable/Drawable;Ljava/lang/String;ILkotlin/jvm/internal/DefaultConstructorMarker;)V in class Lcom/mapbox/maps/plugin/LocationPuck2D; or its super classes (declaration of 'com.mapbox.maps.plugin.LocationPuck2D' appears in /data/app/com.mapbox.navigation.qa_test_app-5MGAb8W4hrQLCkHOpTVAPg==/base.apk!classes29.dex)
    at com.mapbox.androidauto.car.location.CarLocationPuck.navigationPuck2D(CarLocationPuck.kt:10)
    at com.mapbox.androidauto.car.location.CarLocationRenderer.onAttached(CarLocationRenderer.kt:20)
    at com.mapbox.maps.extension.androidauto.CarMapSurfaceOwner.onSurfaceAvailable(CarMapSurfaceOwner.kt:105)
    at androidx.car.app.utils.RemoteUtils$SurfaceCallbackStub.lambda$onSurfaceAvailable$0$androidx-car-app-utils-RemoteUtils$SurfaceCallbackStub(RemoteUtils.java:284)
    at androidx.car.app.utils.RemoteUtils$SurfaceCallbackStub$$ExternalSyntheticLambda5.dispatch(Unknown Source:4)
    at androidx.car.app.utils.RemoteUtils.lambda$dispatchCallFromHost$0(RemoteUtils.java:149)
    at androidx.car.app.utils.RemoteUtils$$ExternalSyntheticLambda2.run(Unknown Source:6)
    at androidx.car.app.utils.ThreadUtils.runOnMain(ThreadUtils.java:39)
    at androidx.car.app.utils.RemoteUtils.dispatchCallFromHost(RemoteUtils.java:146)
    at androidx.car.app.utils.RemoteUtils.lambda$dispatchCallFromHost$1(RemoteUtils.java:185)
    at androidx.car.app.utils.RemoteUtils$$ExternalSyntheticLambda3.run(Unknown Source:8)
    at android.os.Handler.handleCallback(Handler.java:883)
    at android.os.Handler.dispatchMessage(Handler.java:100)
    at android.os.Looper.loop(Looper.java:224)
    at android.app.ActivityThread.main(ActivityThread.java:7590)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
```

**Comment by Zayankovsky** [comment]
Apparently Maps SDK includes two Common versions, 23.1.0-beta.1 and 23.1.0-beta.2. I don't know how compatible they are, but we have a CI check that makes sure all PRE-RELEASE Common SDK versions across dependencies are mismatched. 
https://circleci.com/gh/mapbox/mapbox-navigation-android/108292
```
+--- com.mapbox.maps:android:10.9.0-beta.2
|    +--- com.mapbox.base:common:0.8.0
|    +--- com.mapbox.common:okhttp:23.1.0-beta.2
|    |    \--- com.mapbox.common:common:23.1.0-beta.2 (*)
|    +--- com.mapbox.maps:base:10.9.0-beta.2
|    |    +--- com.mapbox.mapboxsdk:mapbox-android-gestures:0.8.0
|    |    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.5.31 (*)
|    |    +--- com.mapbox.maps:android-core:10.9.0-beta.1
|    |    |    +--- com.mapbox.mapboxsdk:mapbox-sdk-geojson:5.4.1 -> 6.8.0 (*)
|    |    |    +--- com.mapbox.common:common:23.1.0-beta.1 -> 23.1.0-beta.2 (*)
|    |    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.5.31 (*)
|    |    \--- com.mapbox.common:common:23.1.0-beta.2 (*)
|    +--- com.mapbox.extension:maps-style:10.9.0-beta.2
|    +--- com.mapbox.plugin:maps-logo:10.9.0-beta.2
|    +--- com.mapbox.plugin:maps-compass:10.9.0-beta.2
|    +--- com.mapbox.plugin:maps-gestures:10.9.0-beta.2
|    |    \--- com.mapbox.mapboxsdk:mapbox-android-gestures:0.8.0 (*)
|    +--- com.mapbox.plugin:maps-attribution:10.9.0-beta.2
|    +--- com.mapbox.plugin:maps-locationcomponent:10.9.0-beta.2
|    |    \--- com.mapbox.mapboxsdk:mapbox-android-core:5.0.2
|    +--- com.mapbox.plugin:maps-animation:10.9.0-beta.2
|    +--- com.mapbox.plugin:maps-scalebar:10.9.0-beta.2
|    +--- com.mapbox.plugin:maps-overlay:10.9.0-beta.2
|    +--- com.mapbox.plugin:maps-annotation:10.9.0-beta.2
|    |    \--- com.mapbox.extension:maps-style:10.9.0-beta.2
|    +--- com.mapbox.extension:maps-localization:10.9.0-beta.2
|    +--- com.mapbox.plugin:maps-lifecycle:10.9.0-beta.2
|    \--- com.mapbox.plugin:maps-viewport:10.9.0-beta.2
```

File: libnavui-androidauto/build.gradle [hunk 1]
```diff
@@ -40,13 +40,12 @@ dependencies {
 
     // This defines the minimum version of Navigation which is included in this SDK. To upgrade the
     // Navigation versions, you can specify a newer version in your downstream build.gradle.
-    def carNavVersion = "2.9.0-alpha.3"
-    api("com.mapbox.navigation:android:${carNavVersion}")
+    releaseApi("com.mapbox.navigation:android:2.9.0-alpha.3")
```
Comment (kmadsen) [inline libnavui-androidauto/build.gradle:43]: https://github.com/mapbox/mapbox-navigation-android/releases/tag/v2.9.0-alpha.4

^ Still requires `10.8.0`

Comment (Zayankovsky) [reply libnavui-androidauto/build.gradle]: Updated, but we have to keep a separate dependency for debug.

**Comment by Zayankovsky** [comment]
> Apparently Maps SDK includes two Common versions, 23.1.0-beta.1 and 23.1.0-beta.2. I don't know how compatible they are, but we have a CI check that makes sure all PRE-RELEASE Common SDK versions across dependencies are mismatched. https://circleci.com/gh/mapbox/mapbox-navigation-android/108292

@LukasPaczos I see you were the one who updated the check to make sure that all pre-release versions are exactly the same. 23.1.0-beta.2 is compatible with 23.1.0-beta.1, should we relax the check or disable it temporarily?

**Comment by LukasPaczos** [comment]
This is unfortunate. Since this still lands in a relatively early pre-release, let's relax the check as proposed and we'll revisit with the next bump.

**Review by LukasPaczos — APPROVED** [review]

**Comment by LukasPaczos** [comment]
Two changelog entries worth adding:
```
- [TileStoreService] Use shared memory for marshaling large types and improve error handling.
- [TileStoreService] Fixed an issue that prevented callbacks to be invoked if an operation had to be retried because of a loss of connection to the service process.
```

**Comment by Zayankovsky** [comment]
Robo tests are failing because of CORESDK-1344.

**Comment by Zayankovsky** [comment]
> @Zayankovsky could you add a `Known issues` section in the changelog and mention this crash?

#6431

**Pull request merged**